edvart/plots.py

```diff
@@ -68,77 +68,135 @@ def scatter_plot_2d(
         x = df[x]
     if isinstance(y, str):
         y = df[y]
+    plotting_func = _scatter_plot_2d_interactive if interactive else _scatter_plot_2d_noninteractive
+    plotting_func(
+        df=df,
+        x=x,
+        y=y,
+        color_col=color_col,
+        figsize=figsize,
+        opacity=opacity,
+        xlabel=xlabel,
+        ylabel=ylabel,
+        show_xticks=show_xticks,
+        show_yticks=show_yticks,
+        show_zerolines=show_zerolines,
+    )
 
-    if interactive:
-        layout = dict(
-            width=figsize[0] * _INCHES_TO_PIXELS,
-            height=figsize[1] * _INCHES_TO_PIXELS,
-            xaxis=dict(showgrid=False, showticklabels=show_xticks, zeroline=show_zerolines),
-            yaxis=dict(showgrid=False, showticklabels=show_yticks, zeroline=show_zerolines),
-            legend=dict(title=f"<b>{color_col}</b>"),
-        )
 
+def _scatter_plot_2d_noninteractive(
+    df: pd.DataFrame,
+    x: Union[str, pd.Series, np.ndarray],
+    y: Union[str, pd.Series, np.ndarray],
+    color_col: Optional[str] = None,
+    figsize: Tuple[float, float] = (12, 12),
+    opacity: float = 0.8,
+    xlabel: Optional[str] = None,
+    ylabel: Optional[str] = None,
+    show_xticks: bool = False,
+    show_yticks: bool = False,
+    show_zerolines: bool = False,
+) -> None:
+    _fig, ax = plt.subplots(figsize=figsize)
     if color_col is not None:
         is_color_categorical = utils.is_categorical(df[color_col]) or not is_numeric(df[color_col])
```

Comment:
`is_color_categorical` is calculated in both `_scatter_plot_2d_noninteractive` and `_scatter_plot_2d_interactive` -> I'd make it an arg of these functions.

Reply:
Yes, but it's calculated only in some cases (`color_col is not None`). Thus the parameter would often be ignored. I would keep it as is (don't mind duplication of a single line).
I actually noticed it and thought about extracting to a function, but I concluded that it's not worth it for a single LoC.